spec/javascripts/components/story/expanded_story/expanded_story_controls_spec.js

```diff
@@ -55,76 +55,125 @@ describe('<ExpandedStoryControls />', () => {
+      wrapper.find('.cancel').simulate('click');
 
-        expect(button.prop('disabled')).toBe(true);
-      });
+      expect(window.confirm).toHaveBeenCalled();
+    });
 
-      it('disable delete button', () => {
-        const button = wrapper.find('.delete');
+    it('does not trigger a warning window when no changes were made', () => {
+      const handleCancel = sinon.spy();
 
-        expect(button.prop('disabled')).toBe(true);
-      });
+      const wrapper = shallow(
+        <ExpandedStoryControls
+          readOnly={false}
+          isDirty={false}
+          onCancel={handleCancel}
+        />
+      );
 
-      it("don't set disable prop to cancel button", () => {
-        const button = wrapper.find('.cancel');
+      wrapper.find('.cancel').simulate('click');
 
-        expect(button.prop('disabled')).toBe(undefined);
-      });
+      expect(window.confirm).not.toHaveBeenCalled();
     });
 
-    describe("when it's false", () => {
-      let wrapper;
+    describe('when the user clicks cancel on window.confirm', () => {
+      it('does not call onCancel', () => {
+        window.confirm.returns(false);
+        const handleCancel = sinon.spy();
 
-      beforeEach(() => {
-        wrapper = shallow(
-          <ExpandedStoryControls
-            readOnly={false}
-          />
-        );
+        expect(handleCancel).not.toHaveBeenCalled();
       });
+  });
+});
 
-      it("don't disable save button", () => {
-        const button = wrapper.find('.save');
+describe('readOnly', () => {
+  describe("when it's true", () => {
+    let wrapper;
 
-        expect(button.prop('disabled')).toBe(false);
-      });
+    beforeEach(() => {
+      wrapper = shallow(
+        <ExpandedStoryControls
+          readOnly={true}
+        />
+      );
+    });
 
-      it("don't disable delete button", () => {
-        const button = wrapper.find('.delete');
+    it('disable save button', () => {
+      const button = wrapper.find('.save');
 
-        expect(button.prop('disabled')).toBe(false);
-      });
+      expect(button.prop('disabled')).toBe(true);
+    });
+
+    it('disable delete button', () => {
+      const button = wrapper.find('.delete');
+
+      expect(button.prop('disabled')).toBe(true);
+    });
+
+    it("don't set disable prop to cancel button", () => {
+      const button = wrapper.find('.cancel');
+
+      expect(button.prop('disabled')).toBe(undefined);
     });
   });
+
+  describe("when it's false", () => {
+    let wrapper;
+
+    beforeEach(() => {
+      wrapper = shallow(
+        <ExpandedStoryControls
+          readOnly={false}
+        />
+      );
+    });
+
+    it("don't disable save button", () => {
+      const button = wrapper.find('.save');
+
+      expect(button.prop('disabled')).toBe(false);
+    });
+
+    it("don't disable delete button", () => {
+      const button = wrapper.find('.delete');
+
+      expect(button.prop('disabled')).toBe(false);
+    });
+  });
+});
```

Comment:
There's something weird with this indentation, it shouldn't have the same level as the one of the next line. Can you check it?

Reply:
will do.

Reply:
fixed!